lib/system/repr_v2.nim

```diff
@@ -158,7 +158,7 @@ proc repr*[T](x: seq[T]): string =
   ##
   ## .. code-block:: Nim
   ##   $(@[23, 45]) == "@[23, 45]"
-  collectionToRepr(x, "@[", ", ", "]")
```

Comment:
* did you mean x[0] ?
* make sure tests check for case of x.len == 0; but also, see PR comment regarding what address we want to print

Reply:
yes definitely it should be x[0], I wonder why it was compiling anyway... see below for explanation why I think I should change approach.

Reply:
yes, I meant x[0]. will follow up also on address in comment (I do not think my currentapproach works)